### ISSUE TITLE
Translation items in messages.xlf are ordered uncertainly.

### ISSUE THREAD

**Issue by shu-mutou** [angular/angular#41236]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/localize

### Is this a regression?

 Yes, the previous version in which this bug was not present was: @angular/localize": "10.1.6"

### Description

All of several results, i.e. `messages.xlf`, of running `ng extract-i18n` has differences in the order of `<trans-unit>` and `<context>`, in spite of no-change in html files.

`--ivy false` option for `ng extract-i18n` fixes this, but ivy seems default in Angular 11, and our product compiled with ivy. So `ng extract-i18n` shows warnning like follow:

```
Ivy extraction not enabled but application is Ivy enabled. If the extraction fails, the `--ivy` flag will enable Ivy extraction.
```

## 🔬 Minimal Reproduction

1. Run `ng extract-i18n` several times for product compiled with ivy.
2. Run `diff` with `messages.xlf` for each result of `ng extract-i18n`.

## 🔥 Exception or Error
<pre><code>

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 11.2.4
Node: 12.18.4
OS: linux x64
Angular: 11.0.6
... animations, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes
Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.4
@angular-devkit/build-angular   0.1102.4
@angular-devkit/core            11.2.4
@angular-devkit/schematics      11.2.4
@angular/cdk                    11.0.3
@angular/cli                    11.2.4
@angular/flex-layout            11.0.0-beta.33
@angular/language-service       11.2.5
@angular/material               11.0.3
@schematics/angular             11.2.4
@schematics/update              0.1102.4
rxjs                            6.6.6
typescript                      4.0.7
</code></pre>

**Anything else relevant?**

**Comment by shu-mutou** [comment]
This issue makes reviewing changes difficult.

**Comment by shu-mutou** [comment]
I found this issue does not reproduce updating Angular staffs to 11.2.4 or 5.

**Comment by petebacondarwin** [comment]
Strange that you experienced this. A similar bug was reported in https://github.com/angular/angular/issues/40473 and fixed in https://github.com/angular/angular/pull/40192 which was released in v11.1.0.

**Comment by shu-mutou** [comment]
Thank you for your definite information!
Our Angular version was 11.0.6, so we could resolve this issue by updating to 11.2.x.
